src/Firebase/Streaming/FirebaseCache.cs

```diff
@@ -146,7 +146,8 @@ public IEnumerable<FirebaseObject<T>> PushData(string path, string data)
                 }
                 else
                 {
-                    JsonConvert.PopulateObject(data, obj);
+                    var serializerSettings = new JsonSerializerSettings {ObjectCreationHandling = ObjectCreationHandling.Replace};                    
```

Comment:
1) can you move this to constructor & reuse the one instance, so it's not being recreated with every request?
2) can you write a unit test for this scenario?